cryptoki/src/functions/object_management.rs

```diff
@@ -135,7 +135,7 @@ impl<'a> Session<'a> {
     /// let _ = flags.set_rw_session(true).set_serial_session(true);
     ///
     /// let session = pkcs11.open_session_no_callback(slot, flags).unwrap();
-    /// session.login(UserType::User, "fedcba");
+    /// session.login(UserType::User, Some("fedcba"));
```

Comment:
By the way, it seems that if PIN instead of being `&str` would be generic `T` and the parameter was specialized as `where T: Into<Option<&str>` the API could stay the same while allowing `None`:

```rust
fn just_str(s: &str) {
    println!("just s: {}", s);
}

fn into_str<'a, T>(s: T) where T: Into<Option<&'a str>> {
    println!("into s: {:#?}", s.into());
}

fn main() {
    just_str("test");
    
    into_str(Some("test"));
    into_str("test"); // <-- better ergonomics, no need for explicit Some("test")
    into_str(None);
}
```

[Link to playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9ee2a5632ea8e33e6d9781bf0d14e0a0)

Reply:
True, could be an `Option`. Happy for this to be added in a subsequent PR 😄 Didn't know about it!